core/storage/btree.rs

```diff
@@ -1586,6 +1602,17 @@ impl BTreeCursor {
                     cells_capacity_start,
                     "calculation of max cells was wrong"
                 );
+
+                // Let's copy all cells for later checks
+                #[cfg(debug_assertions)]
+                let mut cells_debug = Vec::new();
+                #[cfg(debug_assertions)]
+                {
+                    for cell in &cell_array.cells {
+                        cells_debug.push(cell.to_vec());
+                    }
+                }
```

Comment:
We should hide these checks into `verify_*` functions to make the code more readable.